_posts/2019-11-19-demo-fraud-detection.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: post
+title: "Building a Fraud Detection System with Runtime Rules Updates on Apache Flink"
+date: 2019-11-19T12:00:00.000Z
+authors:
+- alex:
+  name: "Alexander Fedulov"
+  twitter: "afedulov"
+excerpt:
+categories: news
+---
+
+In this series of blog posts you will learn about three powerful Flink patterns for building streaming applications:
+
+ - Dynamic updates of application logic
+ - Dynamic data partitioning (shuffle), controlled at runtime
+ - Low latency alerting based on custom window logic (not using the Window API)
+
+These patterns expand the possibilities of what is achievable with statically defined data flows and thus provide important building blocks to fulfil versatile business requirements.
+
+**Dynamic updates of application logic** allow Flink jobs to change at runtime, without downtime from stopping and resubmitting the code.  
+<br>
+**Dynamic data partitioning** enables the ability to change how events are being distributed and grouped by the `keyBy()` operator in Flink at runtime. Such functionality often becomes a natural requirement when building jobs with dynamically reconfigurable application logic.  
```

Comment:
```suggestion
**Dynamic data partitioning** enables changing how events are being distributed and grouped by the `keyBy()` operator in Flink's runtime. Such functionality often becomes a natural requirement when building jobs with dynamically reconfigurable application logic.  
```

Reply:
This is actually about "at runtime", meaning during the lifecycle of the application, not the "Flin's Runtime", as an execution engine. This is I think "ability to change" might be fine here too.

Reply:
I updated the suggestion based on your comment. Thanks for the input @afedulov